### DISCUSSION TITLE
winget source msstore problem

### DISCUSSION THREAD

**Discussion by poorerhu** [microsoft/winget-cli#4302] [Q&A] [answered]
PS C:\Users\dev> winget upgrade Microsoft.AppInstaller
搜索源时失败: msstore
执行此命令时发生意外错误：
SSL Error: WINHTTP_CALLBACK_STATUS_FLAG_CERT_REV_FAILED failed to check revocation status. WINHTTP_CALLBACK_STATUS_FLAG_INVALID_CA SSL invalid CA.
![image](https://github.com/microsoft/winget-cli/assets/6958217/8113759c-2a21-4e64-b714-21577d441c31)
![屏幕截图 2024-03-28 100234](https://github.com/microsoft/winget-cli/assets/6958217/716154ee-9fff-4515-b442-0ca6321948a0)

 the msstore crashed last year,and I try several methods to fix this problem, finally I give up.
app商店无法打开和winget更新都没用，我用了网上的许多方法，都没法修复，包括勾选ssl tls，重新安装msstore，重置，还是这个鬼样了。

**Comment by Trenly** [comment]
Can you provide the output of `winget --info`

**Reply by poorerhu** [reply]
![屏幕截图 2024-03-28 100028](https://github.com/microsoft/winget-cli/assets/6958217/d75a09f0-2323-4701-922f-5493a0a46037)

**Reply by Trenly** [reply]
Based on the screenshot and the error, it may be something in your network such as a firewall or a proxy which is preventing the SSL certificate from being validated 

> WINHTTP_CALLBACK_STATUS_FLAG_CERT_REV_FAILED -
> Certification revocation checking has been enabled, but the revocation check failed to verify whether a certificate has been revoked. The server used to check for revocation might be unreachable.

**Reply by poorerhu** [reply]
I have no proxy right now. I can open msstore source url in browser ,but it shows:
This XML file does not appear to have any style information associated with it. The document tree is shown below.
<Error>
<script/>
<Message>No HTTP resource was found that matches the request URI 'https://useast.storeedgefd-origin.xbetservices.akadns.net/v9.0'.</Message>
</Error>

**Reply by poorerhu** [reply]
unsafe error when i open 'https://useast.storeedgefd-origin.xbetservices.akadns.net/v9.0', maybe that is the problem, i am in china mainland.-_-

**Comment by denelon** [comment]
You could try:

```PowerShell
$progressPreference = 'silentlyContinue'
Write-Information "Downloading WinGet and its dependencies..."
Invoke-WebRequest -Uri https://aka.ms/getwinget -OutFile Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle
Invoke-WebRequest -Uri https://aka.ms/Microsoft.VCLibs.x64.14.00.Desktop.appx -OutFile Microsoft.VCLibs.x64.14.00.Desktop.appx
Invoke-WebRequest -Uri https://github.com/microsoft/microsoft-ui-xaml/releases/download/v2.8.6/Microsoft.UI.Xaml.2.8.x64.appx -OutFile Microsoft.UI.Xaml.2.8.x64.appx
Add-AppxPackage Microsoft.VCLibs.x64.14.00.Desktop.appx
Add-AppxPackage Microsoft.UI.Xaml.2.8.x64.appx
Add-AppxPackage Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle
```

**Reply by poorerhu** [reply]
I already reinstall them，but did not work

**Comment by poorerhu** [comment, marked as answer]
holy, I just fix it.It's a cert* problem.
use three commands:
> certutil.exe -generateSSTFromWU roots.sst
> $sstStore=(Get-ChildItem -Path roots.sst)
> $sstStore | Import-Certificate -CertStoreLocation Cert:\LocalMachine\Root

参考地址 ref: https://blog.csdn.net/bobodadao/article/details/131855508

**Reply by CaiMiao** [reply]
此方法也可以成功解决该错误
> 尝试更新源失败： winget 打开预定义源失败; 请向 winget 维护人员报告。 执行此命令时发生意外错误： 0x80070005
